app/src/main/java/com/aacademy/homework/ui/movielist/FragmentMoviesList.kt

```diff
@@ -62,37 +57,13 @@ class FragmentMoviesList : Fragment(R.layout.fragment_movies_list) {
         subscribe()
     }
 
-    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
-        super.onCreateOptionsMenu(menu, inflater)
-        val searchItem = menu.findItem(R.id.action_search)
-        val searchView = searchItem?.actionView as SearchView
-        if (viewModel.queryFlow.value.isNotEmpty()) {
-            searchItem.expandActionView()
-            searchView.setQuery(viewModel.queryFlow.value, true)
-            searchView.clearFocus()
-        }
-
-        searchView.setOnQueryTextListener(
-            object : OnQueryTextListener {
-                override fun onQueryTextSubmit(query: String?): Boolean {
-                    lifecycleScope.launch { viewModel.queryFlow.value = (query ?: "") }
-                    return false
-                }
-
-                override fun onQueryTextChange(newText: String?): Boolean {
-                    lifecycleScope.launch { viewModel.queryFlow.value = newText ?: "" }
-                    return false
-                }
-            }
-        )
-    }
-
     private fun initViews() {
         movieAdapter =
             MovieAdapter(
                 Glide.with(this),
                 resources,
                 {
+                    view?.hideKeyboard()
                     (activity as MainActivity).openMovieDetail(it)
```

Comment:
Maybe (activity as MainActivity) like field. This will reduce the number of castes.